drivers/hmis/app/models/hmis/workflow_execution/engine.rb

```diff
@@ -32,6 +32,11 @@ def start_step!(step, user:)
       log_event('start_step', user: user, step: step)
     end
 
+    def validate_step(step, submitted_values:)
+      definition = step.node.form_definition
+      definition.validate_form_values(submitted_values) if definition.present? && submitted_values.present?
+    end
```

Comment:
Not sure about the conditional validation:
- I think task nodes are required to have a definition. We shouldn't need to check it's present unless I'm missing something
- we should validate even if the submitted values were empty, right?

```suggestion
    def validate_step(step, submitted_values:)
      definition = step.node.form_definition
      definition.validate_form_values(submitted_values) 
    end
```

Reply:
Makes sense, thanks!